web/server/codechecker_server/api/report_server.py

```diff
@@ -3424,10 +3424,22 @@ def removeRun(self, run_id, run_filter):
             # cascades and such. Deleting runs in separate transactions don't
             # exceed a potential statement timeout threshold in a DBMS.
             runs = []
+            run_no = 0
+
             for run in q.all():
-                runs.append(run.name)
-                session.delete(run)
-                session.commit()
+                try:
+                    runs.append(run.name)
+                    session.delete(run)
+                    session.commit()
+                    run_no += 1
+                except Exception:
+                    # TODO: Display alert on the GUI if there's an exception
+                    # This exception is handled silently because it is
+                    # expected to never occur, but there have been some rare
+                    # cases where it occurred due to underlying reasons.
+                    # Handling it silently ensures that the Number of runs
+                    # counter is not affected by the exception.
+                    pass
```

Comment:
I appreciate the detailed explanation; silently suppressing an exception can lead to a lot of headache, so whenever we do it, we better make sure we need to.

It just so happens that in my view, we could just emit a simple LOG message _and then_ be done with it. The error occuring rarely is an argument for, not against suppressing this silently. That way, we would have at least SOME idea that something went wrong in here.
```suggestion
                except Exception as e:
                    # TODO: Display alert on the GUI if there's an exception
                    # This exception is handled silently because it is
                    # expected to never occur, but there have been some rare
                    # cases where it occurred due to underlying reasons.
                    # Handling it silently ensures that the Number of runs
                    # counter is not affected by the exception.
                    LOG.warning(f"Suppressed an exception here: {e}")
                    pass
```